query_test.go

```diff
@@ -1,200 +1,247 @@
-	s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 3, 'Klingon' )")
-	s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 4, 'Ferengi' )")
-	s = append(s, "INSERT INTO "+testTableName()+" (id, name, ts) VALUES ( 5, 'Cardassian',"+met+" )")
-	wResults, err = conn.Write(s)
+	s = append(s, "INSERT INTO "+testTableName()+"_full (id, name, wallet, bankrupt, payload, ts) VALUES ( 1, 'Romulan', 123.456, 0, '{\"met\":\""+met+"\"}', "+fmt.Sprint(time.Now().Unix())+" )")
+	s = append(s, "INSERT INTO "+testTableName()+"_full (id, name, wallet, bankrupt, payload, ts) VALUES ( 2, 'Vulcan', 123.456, 0, '{\"met\":\""+met+"\"}', "+fmt.Sprint(time.Now().Unix())+" )")
+	s = append(s, "INSERT INTO "+testTableName()+"_full (id, name, wallet, bankrupt, payload, ts) VALUES ( 3, 'Klingon', 123.456, 1, '{\"met\":\""+met+"\"}', "+fmt.Sprint(time.Now().Unix())+" )")
+	s = append(s, "INSERT INTO "+testTableName()+"_full (id, name, wallet, bankrupt, payload, ts) VALUES ( 4, 'Ferengi', 123.456, 0, '{\"met\":\""+met+"\"}', "+fmt.Sprint(time.Now().Unix())+" )")
+	s = append(s, "INSERT INTO "+testTableName()+"_full (id, name, wallet, bankrupt, payload, ts) VALUES ( 5, 'Cardassian', 123.456, 1, '{\"met\":\""+met+"\"}', "+met+" )")
+	wResults, err := globalConnection.Write(s)
 	if err != nil {
-		t.Logf("--> FATAL")
-		t.Fatal()
+		t.Errorf("failed during insert: %v", err.Error())
+		for _, wr := range wResults {
+			if wr.Err != nil {
+				t.Errorf("caught error: %v", wr.Err.Error())
+			}
+		}
 	}
 
-	t.Logf("trying QueryOne")
-	qr, err = conn.QueryOne("SELECT name, ts FROM " + testTableName() + " WHERE id > 3")
-	if err != nil {
-		t.Logf("--> FAILED (%s)", err.Error())
-		t.Fail()
-	}
+	t.Run("Normal case", func(t *testing.T) {
+		qr, err := globalConnection.QueryOne("SELECT name, ts, wallet, bankrupt, payload FROM " + testTableName() + "_full WHERE id > 3")
+		if err != nil {
+			t.Errorf("failed during query: %v", err.Error())
+		}
 
-	t.Logf("trying Next()")
-	na := qr.Next()
-	if na != true {
-		t.Logf("--> FAILED")
-		t.Fail()
-	}
+		if qr.NumRows() != 2 {
+			t.Errorf("expected 2 row, got %v", qr.NumRows())
+		}
 
-	t.Logf("trying Map()")
-	r, err := qr.Map()
-	if err != nil {
-		t.Logf("--> FAILED (%s)", err.Error())
-		t.Fail()
-	}
-	if r["name"].(string) != "Ferengi" {
-		t.Logf("--> FAILED, expected 'Ferengi', got %s", r["name"].(string))
-		t.Fail()
-	}
-	if ts, ok := r["ts"]; ok {
-		if ts, ok := ts.(time.Time); ok {
-			// time should not be zero because it defaults to current utc time
-			if ts.IsZero() {
-				t.Logf("--> FAILED: time is zero")
-				t.Fail()
-			} else if ts.Before(started) {
-				t.Logf("--> FAILED: time %q is before start %q", ts, started)
-				t.Fail()
+		if len(qr.Columns()) != 5 {
+			t.Errorf("expected 5 columns, got %v", len(qr.Columns()))
+		}
+
+		if len(qr.Columns()) == 5 {
+			expect := []string{"name", "ts", "wallet", "bankrupt", "payload"}
+			for i, c := range qr.Columns() {
```

Comment:
Tip: you can use `assert.ElementsMatch(t, expect, qr.Columns())` function from the `github.com/stretchr/testify/assert` package (not sure if that package exists in this project, but should be easy to add).

Reply:
I don't want to use a third party package for just asserting that one.

Reply:
Ah okay.